migration/tasklist-migration/src/test/java/io/camunda/migration/tasklist/ElasticsearchMigrationRunnerIT.java

```diff
@@ -0,0 +1,245 @@
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+@TestInstance(Lifecycle.PER_CLASS)
+public class ElasticsearchMigrationRunnerIT {
+
+  @Container
+  private static final ElasticsearchContainer ES_CONTAINER =
+      TestSearchContainers.createDefeaultElasticsearchContainer();
+
+  private static ElasticsearchClient esClient;
+  private static MigrationRunner migrator;
+  private static TasklistMigrationProperties properties;
+
+  @BeforeAll
+  public static void setUp() throws IOException {
+    properties = new TasklistMigrationProperties();
+    properties.setBatchSize(5);
+    final ConnectConfiguration connectConfiguration = new ConnectConfiguration();
+    connectConfiguration.setUrl("http://localhost:" + ES_CONTAINER.getMappedPort(9200));
+    properties.setConnect(connectConfiguration);
+    final var connector = new ElasticsearchConnector(properties.getConnect());
+    esClient = connector.createClient();
+    createIndices();
+  }
+
+  private static void createIndices() throws IOException {
+    CreateIndexRequest r =
+        new CreateIndexRequest.Builder()
+            .index(IndexUtil.getSourceIndexName(properties))
+            .withJson(new StringReader(TestData.tasklistIndex()))
+            .build();
+
+    esClient.indices().create(r).acknowledged();
+
+    r =
+        new CreateIndexRequest.Builder()
+            .index(IndexUtil.getTargetIndexName(properties))
+            .withJson(new StringReader(TestData.operateIndex()))
+            .build();
+
+    esClient.indices().create(r).acknowledged();
+  }
+
+  @BeforeEach
+  public void setup() throws IOException {
+    migrator = new MigrationRunner(properties);
+    esClient.deleteByQuery(
+        DeleteByQueryRequest.of(
+            d ->
+                d.index(IndexUtil.getSourceIndexName(properties))
+                    .conflicts(Conflicts.Proceed)
+                    .query(q -> q.matchAll(m -> m))));
+    esClient.deleteByQuery(
+        DeleteByQueryRequest.of(
+            d ->
+                d.index(IndexUtil.getTargetIndexName(properties))
+                    .conflicts(Conflicts.Proceed)
+                    .query(q -> q.matchAll(m -> m))));
+  }
+
+  @Test
+  public void shouldMigrateSuccessfully() throws IOException {
+    // when
+    createEntity(
+        new ProcessEntity()
+            .setId("1")
+            .setVersion(1)
+            .setBpmnXml(TestData.xmlWithForm())
+            .setBpmnProcessId("formProcess"),
+        true);
+    createEntity(
+        new ProcessEntity()
+            .setId("2")
+            .setVersion(1)
+            .setBpmnXml(TestData.xmlWithoutForm())
+            .setBpmnProcessId("formProcess"),
+        true);
+    awaitRecordsArePresent();
+    // then
+    migrator.run();
+
+    // verify
+    assertTargetRecordsAreMigrated();
```

Comment:
🔧 I would inline this method. It makes it easier to follow - to directly see what is asserted in the test.